.ci/Jenkinsfile

```diff
@@ -187,30 +192,87 @@ pipeline {
             }
           }
           steps {
-            withGithubNotify(context: 'E2E Tests', tab: 'tests') {
+            withGithubNotify(context: 'Deploy Stack', tab: 'tests') {
               deleteDir()
               unstash 'source'
               dir("${BASE_DIR}") {
                 script {
-                  def suitesParam = params.runTestsSuites
-                  def existingSuites = readYaml(file: "${TEST_MATRIX_FILE}")
-                  def parallelTasks = [:]
+                  def regexps = [ "^e2e/_suites/fleet/.*", "^e2e/_suites/helm/.*", "^e2e/_suites/kubernetes-autodiscover/.*", "^.ci/.*", "^cli/.*", "^e2e/.*\\.go", "^internal/.*\\.go" ]
+                  def stackWorkspace = "${env.WORKSPACE}/${env.BASE_DIR}"
+                  if ("${FORCE_SKIP_GIT_CHECKS}" == "true" || isGitRegionMatch(patterns: regexps, shouldMatchAll: false)) {
+                    // Deploy the test infrastructure
+                    sh "ssh-keygen -b 4096 -t rsa -f ${stackWorkspace}/e2essh -q -N \"\" "
+                    def stackMachine = getMachineInfo(stackWorkspace, 'stack')
 
-                  if (suitesParam == "") {
-                    log(level: 'DEBUG', text: "Iterate through existing test suites")
-                    existingSuites['SUITES'].each { item ->
-                      checkTestSuite(parallelTasks, item)
-                    }
+                    ansible(stackWorkspace,
+                            env.RUN_ID.split('-')[0],
+                            "-t provision-stack --extra-vars=\"nodeLabel=stack nodeImage=${stackMachine.image} nodeInstanceType=${stackMachine.instance_type}\"")
+
+                    // Must be gathered after deployment as the public IP is known at that time
+                    def stackRunner = getNodeIp(stackWorkspace, 'stack')
+
+                    ansible(stackWorkspace,
+                            env.RUN_ID.split('-')[0],
+                            "-i \"${stackRunner.ip},\" -t setup-stack --extra-vars=\"nodeLabel=stack nodeImage=${stackMachine.image} nodeInstanceType=${stackMachine.instance_type}\"")
+
+                    // Update stash with latest changes from the stack deployment
+                    stash allowEmpty: true, name: 'sourceEnvModified', useDefaultExcludes: false
                   } else {
-                    log(level: 'DEBUG', text: "Iterate through the comma-separated test suites (${suitesParam}), comparing with the existing test suites")
-                    suitesParam.split(',').each { suiteParam ->
-                      existingSuites['SUITES'].findAll { suiteParam.trim() == it.suite }.each { item ->
-                        checkTestSuite(parallelTasks, item)
-                      }
+                    log(level: 'WARN', text: "The test suites won't be executed in any platform because there are no modified files")
```

Comment:
If we are in this else branch, will the next stage (End-To-End Tests) be run? If the else branch is evaluated (both FORCE_SKIP_GIT_CHECKS and isGitRegionMatch are false), then no stack will be provisioned